src/ReverseProxy/Management/ReverseProxyServiceCollectionExtensions.cs

```diff
@@ -143,6 +144,15 @@ public static IReverseProxyBuilder ConfigureHttpClient(this IReverseProxyBuilder
             throw new ArgumentNullException(nameof(configure));
         }
 
+        var service = builder.Services.FirstOrDefault(service => service.ServiceType == typeof(IForwarderHttpClientFactory));
```

Comment:
This only works if your custom service is added first. You'd have the opposite problem if you added it after. Do you have proposals to mitigate that?

Reply:
What if we accept an `IEnumrable<IForwarderHttpClientFactory>` in `ProxyConfigManager` and throw if there's more than 1?

Reply:
> This only works if your custom service is added first. You'd have the opposite problem if you added it after. Do you have proposals to mitigate that?

Yeah, you're right. But it's better than nothing. For this to work if the service is added after, we could implement our own `IServiceCollection` which would be a wrapper to the default one and assign it to `IReverseProxyBuilder`, but I don't like this idea that much. It's way too complicated for this small problem and one could always add the service to the original `IServiceCollection`.

Maybe an Analyzer could be useful too.

> What if we accept an IEnumrable<IForwarderHttpClientFactory> in ProxyConfigManager and throw if there's more than 1?

This wouldn't work, because this code would have add two `IForwarderHttpClientFactory` already. Perhaps `ConfigureHttpClient` could remove the `IForwarderHttpClientFactory` service if it's the default. 
```csharp
ServiceCollection()
            .AddReverseProxy()
            .ConfigureHttpClient((_, _) => { });
```
But it feels weird to check it on `ProxyConfigManager` because it doesn't seems to be this class responsibility.

Reply:
> > What if we accept an IEnumrable in ProxyConfigManager and throw if there's more than 1?

> This wouldn't work, because this code would have add two IForwarderHttpClientFactory already.

How about something like this https://github.com/microsoft/reverse-proxy/commit/c1f6b5048a2ab6e88b94eaa41564f6cb4e616c30?

Reply:
Triage: This is simple enough and provides a usability improvement. We'll take it as is, plus a comment acknowledging the gap.
```suggestion
        // Avoid overriding any other custom factories. This does not handle the case where a IForwarderHttpClientFactory
        // is registered after this call.
        var service = builder.Services.FirstOrDefault(service => service.ServiceType == typeof(IForwarderHttpClientFactory));
```